File: 70-climbing-stairs/climb_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n

        step_one, step_two = 1, 1

        for i in range(n-1):
            temp =   step_one
            step_one = step_one + step_two
            step_two = temp
        
        return step_one

# Example usage:
n = 5
solution = Solution()
print(solution.climbStairs(n))  # Output: 8

File: 70-climbing-stairs/climb_stairs.js
const climbStairs = (n) => {
    if(n<=2) return n;

    const dp = new Array(n+1).fill(0);

    dp[1] = 1;
    dp[2] = 2;

    for(let i=3; i<=n; i++) {
        dp[i] = dp[i-1] + dp[i-2];
    }

    return dp[n];
}

// Example usage:
const n = 5;
console.log(climbStairs(n)); // Output: 8